homeassistant/components/climate/mqtt.py

```diff
@@ -228,14 +216,15 @@ def __init__(self, hass, config, discovery_hash):
     async def discovery_update(self, discovery_payload):
         """Handle updated discovery message."""
         config = PLATFORM_SCHEMA(discovery_payload)
+        self._config = config
         self._setup_from_config(config)
         await self.availability_discovery_update(config)
         await self._subscribe_topics()
         self.async_schedule_update_ha_state()
 
     def _setup_from_config(self, config):
         """(Re)Setup the entity."""
-        self._name = config.get(CONF_NAME)
+        # self._name = config.get(CONF_NAME)
```

Comment:
Leftover?

Reply:
Yes, sorry about that! I'll fix that, thanks for noticing.